pkg/apis/monitoring/v1alpha1/prometheus_operator_types.go

```diff
@@ -26,3 +27,37 @@ type PrometheusOperatorSpec struct {
 	// Image of config-reloader sidecar container
 	ConfigmapReloaderImage string `json:"configmapReloaderImage,omitempty"`
 }
+
+type PrometheusProvisioning struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PrometheusProvisioningSpec `json:"spec"`
+}
+
+// PrometheusProvisioningSpec extends PrometheusSpec from prometheus-operator by adding fields necessary for deployment
+type PrometheusProvisioningSpec struct {
+	prom.PrometheusSpec `json:",inline"`
+	// Namespace to deploy prometheus statefulset in.
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
This should be coming from ObjectMeta, no?

Reply:
Good point. Seems like we won't be needing `Namespace` field in any of the CRDs then.